lib/modules/datasource/pypi/__fixtures__/versions-html-with-periods.html

```diff
@@ -9,4 +9,6 @@ <h1>Links for package.with.periods</h1>
     <a href="">package.with.periods-2.0.1-py3-none-any.whl</a><br>
     <a href="">package_with_periods-2.0.2-py3-none-any.whl</a><br>
     <a href="">package.with.periods-2.0.2.tar.gz</a><br>
+    <a href="">package.with.periods-2.0.3.whl</a><br>
+    <a href="">package.with.periods-2.0.4</a><br>
```

Comment:
This is not obvious to me. Why no version is returned for those entries? Same question applies to: 
https://github.com/renovatebot/renovate/blob/2673867929da63e79468c6a407447e203e9ad733/lib/modules/datasource/pypi/__fixtures__/versions-html-with-non-normalized-name.html#L12-L13

I believe this is coming from this check:
https://github.com/renovatebot/renovate/blob/2673867929da63e79468c6a407447e203e9ad733/lib/modules/datasource/pypi/index.ts#L209

Reply:
The idea is that according to the spec binary (aka wheel distributions) must look like this `{distribution}-{version}(-{build tag})?-{python tag}-{abi tag}-{platform tag}.whl`. So must inlcude a `python tag`, `abi tag` and `platform tag` as well as end with `.whl`

So a package like `package.with.periods-2.0.4` (without `.whl`) or a package like `package.with.periods-2.0.3.whl` without a python-tag is not valid. 

The check is not super strict, so something like `package.with.periods-2.0.3-py3.whl` would be valid, but I think we don't need to be too over specific here.

Reply:
Personally, I would add a test case that states this explicitly. That these entries are not valid and must be not present in the output. I simply overlooked that these are not optional in this long name spec.

Reply:
On another note, are such entries present in real index? I'm no expert here.

Reply:
> On another note, are such entries present in real index? I'm no expert here.

Depends on the implementation, on pypi probably not, but I have seen super weird thing on our internal Artifactory. So if we can filter them out I think that makes sense.